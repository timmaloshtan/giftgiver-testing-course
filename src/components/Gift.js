import React, { Component } from 'react';
import { Form, FormGroup, FormLabel, FormControl, Button } from 'react-bootstrap';

class Gift extends Component {
  constructor(props) {
    super(props);
    this.state = {
      person: '',
      present: '',
    };
  }

  handlePersonChange = (e) => {
    this.setState({
      person: e.target.value,
    });
  }

  handlePresentChange = (e) => {
    this.setState({
      present: e.target.value,
    });
  }

  render() {
    const { removeGift, gift } = this.props;
    return (
      <div className="gift">
        <Form>
          <FormGroup className="gift__form-group">
            <FormLabel>Person</FormLabel>
            <FormControl
              className="input-person"
              onChange={this.handlePersonChange}
            />
          </FormGroup>
          <FormGroup className="gift__form-group">
            <FormLabel>Present</FormLabel>
            <FormControl
              className="input-present"
              onChange={this.handlePresentChange}
            />
          </FormGroup>
        </Form>
        <Button
          className="btn-remove"
          onClick={() => removeGift(gift.id)}
        >
          x
        </Button>
      </div>
    );
  }
}

export default Gift;