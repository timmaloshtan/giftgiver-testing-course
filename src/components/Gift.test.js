import React from 'react';
import { mount } from 'enzyme';

import Gift from './Gift';

describe('Gift component', () => {
  const gift = mount(<Gift />);

  it('should render properly', () => {
    expect(gift).toMatchSnapshot();
  });

  it('should initialize a person and a present in state', () => {
    expect(gift.state()).toEqual({
      person: '',
      present: '',
    });
  });

  describe('when typing into the person input', () => {
    const PERSON = 'Dad';

    beforeEach(() => {
      gift.find('.input-person').at(0).simulate('change', { target: { value: PERSON } })
    });

    it('should update the person in state', () => {
      expect(gift.state().person).toEqual(PERSON);
    })
    
  });
  
  
});
