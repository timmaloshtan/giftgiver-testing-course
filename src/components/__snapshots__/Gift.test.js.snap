// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift component should render properly 1`] = `
<Gift
  gift={
    Object {
      "id": 1,
    }
  }
  removeGift={[MockFunction]}
>
  <div
    className="gift"
  >
    <Form
      inline={false}
    >
      <form
        className=""
      >
        <FormGroup
          className="gift__form-group"
        >
          <div
            className="gift__form-group form-group"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              <label
                className="form-label"
              >
                Person
              </label>
            </FormLabel>
            <FormControl
              className="input-person"
              onChange={[Function]}
            >
              <input
                className="input-person form-control"
                onChange={[Function]}
              />
            </FormControl>
          </div>
        </FormGroup>
        <FormGroup
          className="gift__form-group"
        >
          <div
            className="gift__form-group form-group"
          >
            <FormLabel
              column={false}
              srOnly={false}
            >
              <label
                className="form-label"
              >
                Present
              </label>
            </FormLabel>
            <FormControl
              className="input-present"
              onChange={[Function]}
            >
              <input
                className="input-present form-control"
                onChange={[Function]}
              />
            </FormControl>
          </div>
        </FormGroup>
      </form>
    </Form>
    <Button
      active={false}
      className="btn-remove"
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      <button
        className="btn-remove btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </Button>
  </div>
</Gift>
`;
